c++11/include/opentracing/stringref.h

```diff
@@ -1,6 +1,12 @@
 #ifndef OPENTRACING_STRINGREF_H
```

Comment:
I removed the API functions that std::string_view doesn't have and added a few of the ones from std::string_view that are easy enough to implement. The only exceptions are the constructor taking std::string and operator std::string, which std::string_view doesn't need base std::string has those basic conversions, so it should be possible to swap it out for std::string_view at some point in the future without breaking much.